lib/babe/build_integration_test.go

```diff
@@ -242,77 +242,67 @@ func TestBuildAndApplyExtrinsic(t *testing.T) {
-		0, 0, 0, 0, 216, 5, 113, 87, 87, 40,
-		221, 120, 247, 252, 137, 201, 74, 231,
-		222, 101, 85, 108, 102, 39, 31, 190, 210,
-		14, 215, 124, 19, 160, 180, 203, 54,
-		110, 167, 163, 149, 45, 12, 108, 80,
-		221, 65, 238, 57, 237, 199, 16, 10,
-		33, 185, 8, 244, 184, 243, 139, 5,
-		87, 252, 245, 24, 225, 37, 154, 163, 142}
-	vtx = transaction.NewValidTransaction(types.Extrinsic(txb), &transaction.Validity{})
-	babeService.transactionState.Push(vtx)
+	genesis, genesisTrie, genesisHeader := newWestendDevGenesisWithTrieAndHeader(t)
+	babeService := createTestService(t, ServiceConfig{}, genesis, genesisTrie, genesisHeader)
 
-	zeroHash, err := common.HexToHash("0x00")
+	header := types.NewHeader(genesisHeader.Hash(), common.Hash{}, common.Hash{}, 1, types.NewDigest())
+	bestBlockHash := babeService.blockState.BestBlockHash()
+	rt, err := babeService.blockState.GetRuntime(bestBlockHash)
 	require.NoError(t, err)
 
-	parentHeader := &types.Header{
-		ParentHash: zeroHash,
-	}
+	err = rt.InitializeBlock(header)
+	require.NoError(t, err)
 
-	duration, err := time.ParseDuration("1s")
+	rawMeta, err := rt.Metadata()
+	require.NoError(t, err)
+	var decoded []byte
+	err = scale.Unmarshal(rawMeta, &decoded)
 	require.NoError(t, err)
 
-	slot := Slot{
-		start:    time.Now(),
-		duration: duration,
-		number:   1000,
-	}
+	meta := &ctypes.Metadata{}
+	err = codec.Decode(decoded, meta)
+	require.NoError(t, err)
 
-	bestBlockHash := babeService.blockState.BestBlockHash()
-	rt, err := babeService.blockState.GetRuntime(bestBlockHash)
+	rv := rt.Version()
+	charlie, err := ctypes.NewMultiAddressFromHexAccountID(
+		kr.KeyCharlie.Public().Hex())
 	require.NoError(t, err)
 
-	const authorityIndex uint32 = 0
-	_, err = babeService.buildBlock(parentHeader, slot, rt, authorityIndex, &types.PreRuntimeDigest{})
-	require.NotNil(t, err)
-	require.Equal(t, "cannot build extrinsics: error applying extrinsic: Apply error, type: Payment",
-		err.Error(), "Did not receive expected error text")
+	call, err := ctypes.NewCall(meta, "Balances.transfer", charlie, ctypes.NewUCompactFromUInt(^uint64(0)))
+	require.NoError(t, err)
 
-	txc := babeService.transactionState.(*state.TransactionState).Peek()
-	if !bytes.Equal(txc.Extrinsic, txa) {
-		t.Fatal("did not readd valid transaction to queue")
+	ext := ctypes.NewExtrinsic(call)
+	genHash, err := ctypes.NewHashFromHexString(genesisHeader.Hash().String())
+	require.NoError(t, err)
+
+	so := ctypes.SignatureOptions{
+		BlockHash:          genHash,
+		Era:                ctypes.ExtrinsicEra{IsImmortalEra: true},
+		GenesisHash:        genHash,
+		Nonce:              ctypes.NewUCompactFromUInt(uint64(0)),
+		SpecVersion:        ctypes.U32(rv.SpecVersion),
+		Tip:                ctypes.NewUCompactFromUInt(^uint64(0)),
+		TransactionVersion: ctypes.U32(rv.TransactionVersion),
 	}
+
+	err = ext.Sign(signature.TestKeyringPairAlice, so)
+	require.NoError(t, err)
+
+	extEnc := bytes.Buffer{}
+	encoder := cscale.NewEncoder(&extEnc)
```

Comment:
nit

```suggestion
	extEnc := bytes.NewBuffer(nil)
	encoder := cscale.NewEncoder(extEnc)
```

also replacing `extEnc` as `encodingBuffer` would be nice